### PR TITLE
MDNS in Go: Round 1

### PR DESCRIPTION
**Pull request by TimSimmons**
This is a basic rewrite of `designate-mdns` in Go. It assumes
that you're using `worker-model` style mdns, so no sending
Notifys, no polling. All it does is translate AXFR queries
to database queries, and serialize db rows into DNS protocol.
This is actually really simple, thus the small size, and single
file.

The basic features:
- Listens on TCP/UDP
- Connects to Designate's Database
- Listens for regular DNS queries (A, SOA, etc) and answers to
the best of it's ability.
- Answers AXFR queries for zones.
- Basic config via flags/file
- Makefile

### CODE REVIEW

**Comment by TimSimmons** [comment]
Hey @pglass could I get one last review before I merge this? There's definitely more tests that can be written, but I think this has gone beyond `POC` and is ready for me to deploy to an env with my worker code.

File: mdns_handle_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	"github.com/miekg/dns"
+	"net"
+	"testing"
+
+	"github.com/rackerlabs/mdns"
+)
+
+//{{64796 false 0 false false false false false true false 0} false [{gomdns.com. 252 1}] [] [] [
+//;; OPT PSEUDOSECTION:
+//; EDNS: version 0; flags: ; udp: 4096]}
```
Comment (pglass) [inline mdns_handle_test.go]: you could probably remove this

**Comment by pglass** [comment]
This looks good. You can merge whenever.

**Comment by TimSimmons** [comment]
:shipit: :rocket:

**Pull request merged**